File: src/slang/parse/PrefixOpParselet.java
package slang.parse;

import slang.lex.Token;

public class PrefixOpParselet implements PrefixParselet {
    @Override
    public Expr parse(Parser parser, Token token) {
        Expr operand = parser.expression(Precedence.CALL);
        return new Expr.Prefix(token, operand);
    }
}

File: src/slang/parse/PrefixOpParselet.scala
package slang.parse

import slang.lex.Token

class PrefixOpParselet extends PrefixParselet {
    override def parse(parser: Parser, token: Token): Expr = {
        val operand = parser.expression(Precedence.PREFIX)
        Expr.Prefix(token, operand)
    }
}
